08-translations.Rmd

```diff
@@ -0,0 +1,160 @@
+are menu labels in the GUI. Depending on the version of R that you are using, some of the languages might already
+be available while others may need work. R leverages the [`gettext`](https://www.gnu.org/software/gettext/) program
+to handle the conversion from English to arbitrary target languages.
+
+Having messages available in other languages can be an important bridge for R learners not confident in English --
+rather than learning two things at once (coding in R and processing diagnostic information in English), they can
+focus on coding while getting more natural errors/warnings in their native tongue.
+
+The [`gettext` manual](https://www.gnu.org/software/gettext/manual/index.html) is a more canonical reference for a
+deep understanding of how `gettext` works. This chapter will just give a broad overview, with particular focus on
+how things work for R, with the goal of making it as low-friction as possible for developers and users to contribute
+new/updated translations.
+
+## How translations work
+
+Each of the default packages distributed with R (i.e., those found in `./src/library` such as `base`, `utils`,
+and `stats`) contains a `po` directory that is the central location for cataloguing/translating each package's
+messages.
+
+### .pot files
+
+The `.pot` file is a snapshot of the messages available in a given **domain**. A domain in R typically identifies
+a source package and a source language (either R or C/C++). For example, the file `R-stats.pot`
+(found in the R sources in `./src/library/stats/po`) is a catalogue of all messages produced by R code in the
+`base` package, while `stats.pot` is a catalogue of all messages produced by C code in the `stats` package.
+
+There are two exceptions to the basic pattern described above. The first is the domain for messages produced by
+the C code which is the fundamental backing of R itself (especially, but not exclusively, the C code under
+./src/main). The associated .pot file is R.pot and it is found in `./src/library/base/po`. R-base.pot is a
+normal .pot file because base has a normal `R` directory.
+
+The second is the domain for the Windows R GUI, i.e., the text in the menus and elsewhere in the R GUI program
+available for running R on Windows. These messages are stored in the `RGui.pot` domain, also in the `po`
+directory for `base`, and are most commonly derived from C code found in `./src/gnuwin32`. One reason to keep
+this domain separate is that it is only relevant to one platform (Windows). In particular, Windows has historically
+different character encodings, so that it made more sense for Windows developers to produce translations
+for Windows, since it is non-trivial for non-Windows users to test their translations for the Windows GUI.
+
+#### Generating .pot files
+
+For outside contributors, there's no need to update .pot files -- translators will typically take the R .pot files
```

Comment:
Do you want to consisently use mark-up for .pot and .po files? I.e., `.pot` and `.po`.